### ISSUE TITLE
filesystems: NFS support

### ISSUE THREAD

**Issue by jhvst** [ponkila/homestaking-infra#16]
As title suggest, look into supporting NFS possibly as a host and a client

**Comment by jhvst** [comment]
On my personal configuration, I do NFS and ksmbd mounts, and I have realized that for NFS, there is a big scope creep here: https://lexming.github.io/posts/btrfs-nfs-file-system-loop/

So, when the NFS mount is a btrfs subvolume, it causes problems with NFS exports because NFS does not understand when the same physical disk ID does multiple mounts within itself. In effect, some folders might not appear or are not writable. Debugging this is a PITA and requires in-depth knowledge of NFS host and client mount options.

**Comment by jhvst** [comment]
Currently, the nodes certainly cannot keep up with anything NFS related. We might look into clustered network filesystems later for backing up important files (SeaweedFS, something etcd backed, or syncthing), but closing this for now.

As for examples, my personal config has:
- server: https://github.com/jhvst/nix-config/blob/494cd2dda6ee169b40e9d27e85d7f2abc781aa57/nixosConfigurations/muro/default.nix#L442
- client: https://github.com/jhvst/nix-config/blob/494cd2dda6ee169b40e9d27e85d7f2abc781aa57/nixosConfigurations/starlabs/default.nix#L312

Basically, just follow the NixOS wiki to set it up.
